### PR TITLE
Update/persist to workspace

### PR DESCRIPTION
**Pull request by m477r1x**
# Description
Added note and example of required workflow step needed to persist assets from one job to the next.

# Reasons
I spent several days trying to figure out why assets were being successfully persisted from my first job in the persist step, but then the `attach_workspace` step in my second job was constantly erroring and downloading 0B of assets. Hopefully this note in the documentaion will help others know that this is required in order for `attach_workspace` to work correctly.

### CODE REVIEW

**Review by thomasr8 — APPROVED** [review]

**Comment by devinmarieb** [comment]
@m477r1x We are waiting on confirmation from engineering so I am converting this to draft until we hear back since it was already approved by our team 👍 Thank you!

**Comment by m477r1x** [comment]
@devinmarieb did this get confirmed yet?

**Comment by devinmarieb** [comment]
@m477r1x I'm not sure, I verified with the person who approved this request that he had asked engineering. @thomasr8 is there an update to this?

**Comment by thomasr8** [comment]
@m477r1x Here is an article with a deep dive into workspaces that you might also find useful:

https://circleci.com/blog/deep-diving-into-circleci-workspaces/

Thank You!

**Pull request closed without merging**
